### PR TITLE
DROOLS-4445 : Edit boolean columns in guided decision tables do not support alias with objects

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-4445

https://github.com/kiegroup/appformer/pull/776
https://github.com/kiegroup/drools-wb/pull/1218

### CODE REVIEW

**Comment by Rikkola** [comment]
@jomarko Adding tests for this, but the fix is in and if you want the review can start.

**Comment by Rikkola** [comment]
@karreiro Hi, perhaps you could help with the review?

**Comment by Rikkola** [comment]
> [ERROR] Failed to execute goal org.kie:kie-maven-plugin:7.26.0-SNAPSHOT:build (default-build) on project mortgage-process: Execution default-build of goal org.kie:kie-maven-plugin:7.26.0-SNAPSHOT:build failed: Unable to get KieModule, Errors Existed: Error Messages:
[ERROR] Message [id=1, kieBase=defaultKieBase, level=ERROR, path=com/myspace/mortgage_app/MortgageApprovalProcess.bpmn, line=-1, column=0
[ERROR]    text=Process Compilation error The method getProperty() is undefined for the type Object]
[ERROR] ---
[ERROR] Warning Messages:
[ERROR] ---
[ERROR] Info Messages:
[ERROR] -> [Help 1]

jenkins please retest this

File: drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/ConstraintValueEditor.java [hunk 1]
```diff
@@ -726,29 +722,29 @@ public void setOnTemplateValueChangeCommand(Command onTemplateValueChangeCommand
...
+        }
 
-            this.dropDownData = oracle.getEnums(this.factType,
-                                                fieldName,
-                                                currentValueMap);
+        this.dropDownData = oracle.getEnums(this.factType,
+                                            fieldName,
+                                            currentValueMap);
+
+        if (DataType.TYPE_BOOLEAN.equals(this.fieldType) && this.dropDownData != null) {
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/ConstraintValueEditor.java]: Is this `if` block correct? To me it seems we override declared enums exactly for case, customer is complaining about in ticket.

**Comment by jomarko** [comment]
### Manual review
Passed.
However I would like to understand `ConstraintValueEditor.initDropDownData` before approving and merge, so I am able to test in community instead of selenium.

File: drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/ConstraintValueEditor.java (hunk 1 above)
Comment (Rikkola) [reply drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/ConstraintValueEditor.java]: Nope it is not. It should be `this.dropDownData == null`. I'll get it fixed.

Comment (Rikkola) [reply drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/ConstraintValueEditor.java]: Fixed. Well spotted. I actually fixed this when I was about to write a test for it, but had to give up. To get a test workin I might have to edit this part more than I wanted and break it again.
The code affects enumerations for booleans in the Guided Rule Editor.

Comment (Rikkola) [reply drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/ConstraintValueEditor.java]: If we decide this needs tests or changes, I would rather revert changes in this file and make a separate ticket for GRE that we fix later.

**Comment by jomarko** [comment]
@Rikkola thank you for updates, please let me know when it is ready for a re-review.

**Comment by Rikkola** [comment]
@jomarko and this one is ready for rereview.

**Comment by jomarko** [comment]
### Manual review
Passed. @Rikkola please have a look on my coverage increase and I will approve.

**Comment by Rikkola** [comment]
@kiegroup/gatekeepers Can I please have these PRs merged thank you.

**Pull request merged**